src/components/PostDetailsPage/EditPostDetailPage.tsx

```diff
@@ -0,0 +1,151 @@
+import { useParams, useNavigate } from 'react-router-dom';
+import { useGetJobsDetailQuery, useUpdateJobPostMutation } from 'service/httpService';
+import { useTranslation } from 'react-i18next';
+import {
+	Container,
+	Title,
+	CreateButton,
+	Column,
+	Li,
+	CurrencyColumn,
+	P,
+} from '../jobPost/JobPost.styles';
+import React, { FC } from 'react';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import ValidationSchema from './validationSchema';
+import { EditJobSubmitForm } from 'components/jobPost/interfaces';
+
+const EditPostDetailPage: FC = () => {
+	const [updatejobPost, {}] = useUpdateJobPostMutation();
+	const { t } = useTranslation();
+	const params = useParams();
+	const { data: post } = useGetJobsDetailQuery(params.id);
+	const postId = post.id;
+	const navigate = useNavigate();
+
+	const {
+		register,
+		handleSubmit,
+		formState: { errors },
+	} = useForm<EditJobSubmitForm>({
+		resolver: yupResolver(ValidationSchema),
+	});
+
+	const onSubmit = async (data: EditJobSubmitForm) => {
+		await updatejobPost({ data, postId });
+		navigate(`/post-job/${params.id}`);
+	};
+	return (
+		<Container>
+			<form onSubmit={handleSubmit(onSubmit)}>
+				<div>
+					<Title>{`${t('EditPostDetailPage.title')}`}</Title>
+					<input
```

Comment:
I thought that I don’t need to change someone else’s code and put it into a separate component, because on my page, these inputs are still needed with defaultValue, that I select from the selected post.